scipy/spatial/distance.py

```diff
@@ -1837,7 +1835,7 @@ def _cosine_cdist(XA, XB, dm):
     dm += 1
 
 
-def cdist(XA, XB, metric='euclidean', p=2, V=None, VI=None, w=None):
+def cdist(XA, XB, metric='euclidean', p=None, w=None, V=None, VI=None):
```

Comment:
You can't change the order of kwargs, that breaks backwards compatibility. ``w`` should stay at the end.